run/logging-manual/test/system.test.js

```diff
@@ -0,0 +1,114 @@
+  let entries;
+
+  for (i = 0; i < MAX_ATTEMPTS; i++) {
+    await setTimeoutPromise(WRITE_CONSISTENCY_DELAY_MS);
+    entries = await getLogEntries(filter);
+    if (entries[0] && entries[0].length > 0) {
+      return entries[0];
+    }
+  }
+
+  return [];
+};
+
+const getLogEntries = async (filter) => {
+  const preparedFilter = `resource.type="cloud_run_revision" ${filter}`;
+  let entries = await logging.getEntries({
+    filter: preparedFilter,
+    autoPaginate: false,
+    pageSize: 2
+  });
+
+  return entries;
+}
+
+const dateMinutesAgo = (date, min_ago) => {
+  date.setMinutes(date.getMinutes() - min_ago)
+  return date.toISOString();
+}
+
+describe('Logging', () => {
+
+  let requestLog;
+  let sampleLog;
+
+  describe("Live Service", () => {
+    it ('can be reached by an HTTP request', async () => {
+      const url = process.env.CLOUD_RUN_SERVICE_URL;
+      if (!url) {
+        throw Error('"$CLOUD_RUN_SERVICE_URL" environment variable is required. For example: https://service-x8xabcdefg-uc.a.run.app.')
+      }
+      await request(url);
+    });
+
+    it ('generates Stackdriver logs', async() => {
+      // The latest two log entries for our service in the last 5 minutes
+      // are treated as the result of the previous test.
+      // Concurrency is supporting by distinctly named service deployment per test run.
+      let filter = `resource.labels.service_name="${service_name}" timestamp>="${dateMinutesAgo((new Date), 5)}"`;
+      const entries = await getLogEntriesPolling(filter);
+      entries.forEach(entry => {
+        if (entry.metadata.httpRequest) {
+          requestLog = entry;
+        } else {
+          sampleLog = entry;
+        }
+      });
+      assert(entries.length >= 2, 'creates at least 2 log entries per request');
+    });
+
+  });
+
+  describe('Structured Logging', () => {
+
+    it('retains "message" property for display text', () => {
+      assert(sampleLog.data.message, "property found in the log entry");
+    });
+
+    it('retains structure for arbitrary properties', () => {
+      assert(sampleLog.data.component, "property found in the log entry");
+    });
+
+    it('uses "severity" property as entry severity', () => {
+      assert(sampleLog.metadata.severity, "'severity' lifted to official property");
+    });
+
+    it('uses "trace" property as valid entry trace ID', () => {
+      assert(sampleLog.metadata.trace, "'trace' lifted to official property");
+      assert(sampleLog.metadata.trace == requestLog.metadata.trace, "`trace` property matches request log trace ID");
+    });
+  });
+
+  describe('Cloud Run Log Metadata', () => {
+    it('has expected label properties', () => {
+      let labels = sampleLog.metadata.labels;
+      let resource = sampleLog.metadata.resource;
+
+      assert(labels.instanceId, "'labels.instanceId' found in the log entry");
+      assert(resource.labels.configuration_name, "'resource.labels.configuration_name' found in the log entry");
```

Comment:
Is it worth shortening the assertion descriptions + doing this in a `for` loop instead?

e.g:
```javascript
const desiredProps = ['configuration_name', 'location', 'revision_name', ...]
desiredProps.each((prop) => {
  assert(resource.labels[prop], `${prop} found in log entry`
});
```

Reply:
Here's what that would look like in practice, dynamic is nice but I'm not sure it's all that much clearer:

```
      const {labels, resource} = sampleLog.metadata;
      const labelProps = ['instanceId']

      labelProps.forEach(prop => {
        assert(labels[prop], `${prop} found in log entry`)
      })

      const resourceProps= [
        'configuration_name',
        'location',
        'project_id',
        'revision_name',
        'service_name'
      ];
      resourceProps.forEach(prop => {
        assert(resource.labels[prop], `${prop} found in log entry`)
      });
```